projects/plugins/boost/app/modules/cache/storage/File_Storage.php

```diff
@@ -19,7 +19,7 @@ class File_Storage implements Storage {
 	private $root_path;
 
 	public function __construct( $root_path ) {
-		$this->root_path = Boost_Cache_Utils::trailingslashit( $root_path );
+		$this->root_path = Boost_Cache_Utils::sanitize_file_path( Boost_Cache_Utils::trailingslashit( $root_path ) );
```

Comment:
🤔 Is it a good idea to sanitize the root path like you would a URL?

It means that any `..` and any `\` gets removed. Wouldn't that break on a Windows system?

Perhaps we need to start "normalizing" the request_uri to include the hostname at the Boost_Cache level? Would that be weird? Would that cause weirdness elsewhere in the system?

This kind of path handling has so many weird edge cases to think about :)

Reply:
It probably won't do anything bad, except for the "\" which some Windows hosts might use. I think it might make sense not to pass WP_CONTENT_DIR to File_Storage. Have it added in the storage class instead.

I'll give this a better test using a test site I have that isn't running at "/" too.

Reply:
I decided to change it so it only passed $home to File_Storage. Let the File_Storage class define where in the filesystem it should store the cache files.